build.rs

```diff
@@ -1,11 +1,55 @@
 extern crate bindgen;
 
+use std::env;
+use std::io;
 use std::path::PathBuf;
 use std::process::Command;
 
 const HEADER_PATH: &str = "include/uapi/linux/nitro_enclaves.h";
 const OUT_FILE: &str = "/driver_structs.rs";
 
+// Generate the sysroot path based on the architecture the program is built.
+// This is needed for the compiler to locate the architecture dependent
+// header files.
+fn get_gcc_sysroot() -> String {
+    // Initialize the output because it might remain uninitialized with the
+    // current control flow
+    let mut output = Err(io::Error::from(io::ErrorKind::NotFound));
+    let mut cc_found = false;
+
+    // Use the `CC` environment variable if existing
+    // which should give us directly the compiler's name
+    let cc = env::var("CC");
+    if let Ok(cc) = cc {
+        output = Command::new(cc).arg("--print-sysroot").output();
+        if let Ok(_) = output {
+            cc_found = true;
+        }
+    }
+
+    if !cc_found {
+        // Otherwise, try to guess the compiler's name by the name
+        // $TARGET-gcc, where $TARGET represents the target architecture
+        // that was specified for build
+        let target = env::var("TARGET").unwrap();
+        output = Command::new(format!("{}-gcc", target))
+            .arg("--print-sysroot")
+            .output();
+        if let Err(_) = output {
+            // If that did not work as well, use directly `gcc`
+            output = Command::new("gcc").arg("--print-sysroot").output();
+        }
+    }
+
+    // At this point we should have a valid command output
+    // that gives us the sysroot path
+    let output = output.unwrap();
```

Comment:
it should be an expect with a helpfull error message so that the user knows what to do :D.

Reply:
That could be helpful indeed. Will change.